MainProgramTests/WildcardTests.cs

```diff
@@ -0,0 +1,46 @@
+﻿using NUnit.Framework;
+using NUnit.Framework.Interfaces;
+using Logic;
+
+namespace MainProgramTests;
+
+internal class WildcardTests
+{
+    private static bool HasPreviousTestPassed = true;
+
+    [SetUp]
+    public void SetUp()
+    {
+        Assume.That(HasPreviousTestPassed, Is.True);
+    }
+
+    [TearDown]
+    public void TearDown()
+    {
+        HasPreviousTestPassed = 
+            HasPreviousTestPassed && TestContext.CurrentContext.Result.Outcome.Status == TestStatus.Passed;
+    }
+
+    [Test, Order(1)]
+    public void ConvertingTest()
+    {
+        var regexPattern = @".*\.youtube\.com/.*";
+        var wildcard = @"*.youtube.com/*";
+        var wildcardRegex = WildcardUtilities.WildcardToRegex(wildcard);
+        Assert.AreEqual(regexPattern, wildcardRegex.ToString());
+    }
+
+    [Test, Order(2)]
+    public void MatchTest()
```

Comment:
It is better to use TestCase here: https://docs.nunit.org/articles/nunit/writing-tests/attributes/testcase.html

And Orders are not required

Reply:
Resolved in new commit